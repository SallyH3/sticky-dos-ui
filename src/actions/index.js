export const setCardList = cardList =>{
         console.log("VIP List", cardList)
         return {
           type: "SET_CARDLIST",
           cardList
         }};

export const deleteCard = (id) => ({
	type: 'DELETE_CARD',
	id
})