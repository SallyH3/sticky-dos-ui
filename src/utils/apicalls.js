export const fetchCardList = (url) => {
	return fetch(url)
	.then(response => {
		if(!response.ok) {
			throw Error('Error fetching data')
		} else {
			return response.json();
		}
	})
}

export const postFetch = (url, init) => {
	return fetch(url, init)
	.then(response => {
		if(!response.ok) {
			throw Error('Error posting data')
		} else {
			return response;
		}
	})
}