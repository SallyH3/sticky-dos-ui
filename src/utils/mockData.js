 export const mockCardList = [
  { 
    id: 0, 
    title: "Welcome to Sticky Do's", 
    content: [
      {
        id: 1,
        type: 'note',
        text: 'This is a standard note. Which is the default when you start typing in our input box ⤴️',
        checked: null
      }
    ]
  },
  { 
    id: 2, 
    title: "Example #2", 
    content: [
      {
        id: 1,
        type: 'list',
        text: 'This is a list item. If you click the ☑️ button in the input box',
        checked: false
      },
      {
        id: 2,
        type: 'list',
        text: 'You can add as many checkList items as you would like!',
        checked: true
      }
    ]
  }
]