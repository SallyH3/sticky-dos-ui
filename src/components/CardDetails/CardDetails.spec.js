import React from 'react';
import { shallow } from 'enzyme';
import { CardDetails, mapDispatchToProps } from './index.js';
import { deleteCard, updateCard } from '../../actions';
import { mockCardList } from '../../utils/mockData';
import { dynamicFetch } from '../../utils/apicalls.js';

let mockCard = mockCardList[1];
let mockState = {
  redirect: false,
  updatedCard: {
    title: mockCard.title,
    id: mockCard.id,
    content: mockCard.content       
	}
}
let handleDeleteCard = jest.fn()

describe('CardDetails', () => {
	let { title, content, id } = mockCard
	let mockDeleteCard;
  let wrapper;
  let expectedCard;


  beforeEach(() => {
  	expectedCard = mockState.updatedCard
  	mockDeleteCard = jest.fn()
    wrapper = shallow(
      <CardDetails 
      	title={title}
      	content={content}
      	id={id}
      	deleteCard={mockDeleteCard}
      	updateCard={updateCard}
      />
    )
   

  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should handleClick', () => {
  	expect(wrapper.state("redirect")).toEqual(false);
  	wrapper.find(".Card__trash").simulate('click');
  	expect(wrapper.state("redirect")).toEqual(true);
  });

  it('should update title state on input', () => {
  	let mockEvent = {
  		target: {
  			value: 'hello there'
  		}
  	};
  	expect(wrapper.state()).toEqual(mockState);
  	wrapper.instance().handleTitleChange(mockEvent);
  	expect(wrapper.state()).toEqual({
  		redirect: false,
 			updatedCard: {
    		title: 'hello there',
    		id: mockCard.id,
    		content: mockCard.content       
			}
		});
  });

  it('should update checkbox state on click', () => {
  	expect(wrapper.state()).toEqual(mockState);
  	wrapper.find("#check1").simulate('click');
  	wrapper.instance().handleCheck(content[0]);
  	expectedCard.content[0].checked= true;
  	expect(wrapper.state()).toEqual({
  		redirect: false,
 			updatedCard: expectedCard
		});
  });

  it('should be able to update li item', () => {
  	let mockEvent = {
  		target: {
  			value: 'hello world'
  		}
  	}
  	expectedCard.content[0].text = 'hello world';
  	expect(wrapper.state()).toEqual(mockState);
  	wrapper.instance().handleLIChange(mockEvent, content[0]); 
  	expect(wrapper.state()).toEqual({
  		redirect: false,
 			updatedCard: expectedCard
		});
  });

  it.skip('should delete card if it matches id', () => {
  	// wrapper.find(".Card__trash").simulate('click');
  	wrapper.instance().handleDeleteCard();
  	expect(deleteCard).toHaveBeenCalled();
  });

  it('should handle save', () => {
    const mockEvent = { target: {} }
    const dynamicFetch = jest.fn();

    expectedCard.content[0].text = 'hello world';
    expect(wrapper.state()).toEqual(mockState);
    wrapper.instance().handleSave(
     { preventDefault: () => {} } , content[0]);
    expect(dynamicFetch).toHaveBeenCalled();
  });

    describe('mapDispatchToProps', () => {
			const mockDispatch = jest.fn();
    	it('should call delete card dispatch when using a function from MDTP', () => {
      
      const actionToDispatch = deleteCard(0);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteCard(0);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    	it('should call update card dispatch when using a function from MDTP', () => {
    		const mockEvent = {
    			target: {
    				value: 'hello world'
    			}
    		};

    		expectedCard.content[0].text = 'hello world';
			  expect(wrapper.state()).toEqual(mockState);
		  	wrapper.instance().handleLIChange(mockEvent, content[0]); 
		  	expect(wrapper.state()).toEqual({
		  		redirect: false,
		 			updatedCard: expectedCard
				});
    		const actionToDispatch = updateCard(expectedCard);
    		const mappedProps = mapDispatchToProps(mockDispatch);
    		mappedProps.updateCard(expectedCard);

    		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    	});
  });

});