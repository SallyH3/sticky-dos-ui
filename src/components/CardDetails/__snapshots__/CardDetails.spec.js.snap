// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should match snapshot 1`] = `
<article
  className="big-card"
>
  <form>
    <fieldset
      className="Card__header"
    >
      <input
        className="title"
        onChange={[Function]}
        placeholder="Example #2"
        type="text"
        value="Example #2"
      />
      <input
        className="Card__save"
        onClick={[Function]}
        type="submit"
      />
      <button
        className="Card__trash"
        onClick={[Function]}
      >
        <i
          class="far fa-trash-alt"
        />
      </button>
    </fieldset>
    <fieldset
      className="content"
    >
      <fieldset
        className="f-li"
        key="1"
      >
        <i
          className="far check-btn  fa-square"
          id="check1"
          onClick={[Function]}
        />
        <input
          id="input1"
          onChange={[Function]}
          type="text"
          value="This is a list item. If you click the ☑️ button in the input box"
        />
      </fieldset>
      <fieldset
        className="f-li"
        key="2"
      >
        <i
          className="fas check-btn fa-check-square"
          id="check2"
          onClick={[Function]}
        />
        <input
          id="input2"
          onChange={[Function]}
          type="text"
          value="You can add as many checkList items as you would like!"
        />
      </fieldset>
    </fieldset>
  </form>
</article>
`;
